src/PermissionMatcher.php

```diff
@@ -12,33 +12,42 @@ class PermissionMatcher
      *
      * @param array|object $permissions
      * @param array        $authzGroups
+     * @param boolean      $reverse     Returns an array where the priviliges are the keys and authzgroups the values
      * @return array
      */
-    public function match($permissions, array $authzGroups)
+    public function match($permissions, array $authzGroups, $reverse = false)
```

Comment:
Changing the meaning of the output of a function by setting a (random) argument to `true` is confusing. _This is actually one of the main problems developers have with PHP, which does that in it's core libraries._

Instead add a function `reverseMatch($permissions, $authzGroups)` (or `matchReverse` if you prefer).